### PR TITLE
[verilator, tests] Fix and tag failing & timed-out Verilator test targets

### PR DESCRIPTION
**Pull request by AlexJones0**
This PR partially addresses issues #24182 and #24184.

All Verilator test targets were run using public CI overnight in preparation for the addition of a weekly/nightly Verilator job in PR #24145. Of the 390 targets (at the time of testing), 14 explicitly failed and 67 timed out (when using `--test_timeout=2400,2400,3600,-1`). This PR accordingly:

- Marks all 67 timed-out tests as "manual", as they take at least 40 (and many > 60) minutes to complete when run on our public CI infrastructure and as such are too long to run in any sort of regular CI. The confirmation of whether these tests truly take a long time or should be marked broken and/or fixed is tracked by #24184. All of these tags are documented with the reason for their inclusion, mentioning whether the timeout used was 40 or 60 minutes for future reference.
- Marks all 6 failing Verilator tests tracked in #24182 as `broken`, until they can be looked at further.
- Fixes the `example_mem_ujcmd_test` test for Verilator, which was incorrectly passing in the Firmware ELF as a keyword argument instead of a positional argument. This has been verified (`PASSED in 599.6s`) by running 
```
./bazelisk.sh test -t- --test_output=streamed --test_timeout=3600,3600,3600,3600 \
    //sw/device/tests:example_mem_ujcmd_test_sim_verilator
```
- Fixes the `spi_host_irq_test` target, which was failing on Verilator due to the SPI clock frequency being configured to 1 MHz, whereas Verilator's HiSpeed peripheral frequency is set to just 500 KHz. A conditional check is added and the SPI clock speed is reduced to 100 KHz if running on Verilator. This has been verified (`PASSED in 133.4s`) by running 
```
./bazelisk.sh test -t- --test_output=streamed --test_timeout=3600,3600,3600,3600 \
    //sw/device/tests:spi_host_irq_test_sim_verilator
```
- Correctly excludes the Verilator execution environment from `gpio_smoketest`, which previously erroneously had the `//hw/top_earlgrey:verilator` environment excluded instead of the correct `//hw/top_earlgrey:sim_verilator`. This occurrence is fixed in 3 other places in the `BUILD` file.
- Removes both the `sim_dv` and `sim_verilator` execution environments from the `usbdev_aon_wake_reset_test` Bazel test target, as [the test source already immediately fails if the device type is DV or Verilator](https://github.com/lowRISC/opentitan/blob/70142638925f66eae20f6232b5a7c1ffc51ff0e3/sw/device/tests/usbdev_aon_wake_reset_test.c#L120) - this lack of support for simulator environments is now made explicit in the Bazel parameters.
- Marks the `spi_host_winbond_flash_test` as being `manual` for the Verilator environment, as it requires access to Winbond flash, which is not currently available in CI.
- Removes the Verilator execution environment from `rv_core_ibex_mem_test_prod`, because [the `rv_core_ibex` tests are marked as being incompatible with test_rom](https://github.com/lowRISC/opentitan/blob/70142638925f66eae20f6232b5a7c1ffc51ff0e3/sw/device/tests/BUILD#L5756-L5757), which Verilator uses, and as such this environement should not be supported.
- Marks the `pwrmgr_all_reset_reqs_test` Verilator execution environment as broken alongside the existing broken `fpga_cw310_sival_rom_ext` environment, as the test provides the same error in both cases.
- Removes the Verilator execution environment from `spi_host_smoketest`, as that test requires a flash connected to the SPI bus which is not emulated by Verilator.

Aside from the few fixes, this PR only marks Verilator tests as manual or broken (or removes the Verilator execution environment). Existing Verilator tests that are running in CI are unaffected by this, as currently only `aes_smoketest` is being run on [English breakfast](https://github.com/lowRISC/opentitan/blob/70142638925f66eae20f6232b5a7c1ffc51ff0e3/ci/scripts/run-english-breakfast-verilator-tests.sh#L37), and [16 specific quick tests are being run on top_earlgrey](https://github.com/lowRISC/opentitan/blob/70142638925f66eae20f6232b5a7c1ffc51ff0e3/ci/scripts/run-verilator-tests.sh#L19C2-L34C64).

### CODE REVIEW

**Review by jwnrt — APPROVED** [review]
Thanks, this LGTM if it passes CI. Thanks also for commenting the reasons why they're broken / manual.

**Comment by AlexJones0** [comment]
CI failed because I didn't autogenerate the `plic` tests with the added manual verilator tag and comments. I've just pushed that change so hopefully CI should pass now with that

**Comment by AlexJones0** [comment]
CI failed because I didn't autogenerate the `plic` tests with the added manual verilator tag and comments, and because of two instances of failed Bazel BUILD linting. I've just pushed those changes so hopefully CI should pass now with that.

File: sw/device/tests/BUILD
Comment (engdoreis) [inline sw/device/tests/BUILD:3445]: Why manual? Isn't this test that needs a spi flash on the bus and therefore can't be tests in verilator?

**Review by engdoreis — CHANGES_REQUESTED** [review]
Overall it LGTM, thanks @AlexJones0 for the good work, I suggested some changes below.
I'm also blocking this PR with the "Request Changes" to avoid it merge before the end of the quiet period.

**Comment by AlexJones0** [comment]
Now that quiet period has ended, I've rebased on the latest changes to `master`, and I've also addressed the remaining feedback on the comments, code style and `spi_host_winbond_flash_test`. 

Everything from the original PR should still be relevant, except for the comment on `spi_host_winbond_flash_test`. Rather than marking this as `manual` for the Verilator environment, the Verilator sim environment is now removed from this test entirely, as a result of the test requiring an SPI flash on the bus, which is not available in Verilator.

**Review by engdoreis — APPROVED** [review]

**Pull request merged**
